packages/ra-supabase-ui-materialui/src/guessers/CreateGuesser.tsx

```diff
@@ -0,0 +1,127 @@
+    InferredElement,
+    editFieldTypes,
+} from 'react-admin';
+import type { CreateProps, CreateViewProps } from 'react-admin';
+import { capitalize, singularize } from 'inflection';
+
+import { inferElementFromType } from './inferElementFromType';
+
+export const CreateGuesser = (props: CreateProps & { enableLog?: boolean }) => {
+    const {
+        mutationOptions,
+        resource,
+        record,
+        transform,
+        redirect,
+        disableAuthentication,
+        ...rest
+    } = props;
+    return (
+        <CreateBase
+            resource={resource}
+            record={record}
+            redirect={redirect}
+            transform={transform}
+            mutationOptions={mutationOptions}
+            disableAuthentication={disableAuthentication}
+        >
+            <CreateGuesserView {...rest} />
+        </CreateBase>
+    );
+};
+
+export const CreateGuesserView = (
+    props: CreateViewProps & {
+        enableLog?: boolean;
+    }
+) => {
+    const { data: schema, error, isPending } = useAPISchema();
+    const resource = useResourceContext();
+    const [child, setChild] = React.useState<ReactNode>(null);
+    if (!resource) {
+        throw new Error('CreateGuesser must be used withing a ResourceContext');
+    }
+    const { enableLog = process.env.NODE_ENV === 'development', ...rest } =
+        props;
+
+    React.useEffect(() => {
+        if (isPending || error) {
+            return;
+        }
+        const resourceDefinition = schema.definitions?.[resource];
+        const requiredFields = resourceDefinition?.required || [];
+        if (!resourceDefinition || !resourceDefinition.properties) {
+            throw new Error(
+                `The resource ${resource} is not defined in the API schema`
+            );
+        }
+        if (!resourceDefinition || !resourceDefinition.properties) {
+            return;
+        }
+        const inferredInputs = Object.keys(resourceDefinition.properties)
+            .filter((source: string) => source !== 'id')
+            .map((source: string) =>
+                inferElementFromType({
+                    name: source,
+                    types: editFieldTypes,
+                    description:
+                        resourceDefinition.properties![source].description,
+                    format: resourceDefinition.properties![source].format,
+                    type: (resourceDefinition.properties &&
+                    resourceDefinition.properties[source] &&
+                    typeof resourceDefinition.properties[source].type ===
+                        'string'
+                        ? resourceDefinition.properties![source].type
+                        : 'string') as string,
+                    requiredFields,
+                })
+            );
+        const inferredForm = new InferredElement(
+            editFieldTypes.form,
+            null,
+            inferredInputs
+        );
+        setChild(inferredForm.getElement());
```

Comment:
This should probably be a hook reusable between views

Reply:
I disagree. This refactoring is not straightforward as we can only mutualize part of the effect (the console.log is different), and it has to interface with a state defined outside. I assume the refactoring you suggest will produce code that is harder to read (and the current code is already hard to read).